tests/autologging/test_autologging_safety_unit.py

```diff
@@ -1320,15 +1382,55 @@ def test_validate_args_throws_when_arg_types_or_values_are_changed():
 
     with pytest.raises(Exception, match="does not match expected type"):
         _validate_args(
-            user_call_args, user_call_kwargs, invalid_autologging_call_args_2, user_call_kwargs
+            "",
+            "",
+            user_call_args,
+            user_call_kwargs,
+            invalid_autologging_call_args_2,
+            user_call_kwargs,
         )
 
     with pytest.raises(Exception, match="does not match expected type"):
         _validate_args(
-            user_call_args, user_call_kwargs, user_call_args, invalid_autologging_call_kwargs_2
+            "",
+            "",
+            user_call_args,
+            user_call_kwargs,
+            user_call_args,
+            invalid_autologging_call_kwargs_2,
         )
 
 
+@pytest.mark.usefixtures(test_mode_on.__name__)
+def test_validate_args_exempts_tensorflow_keras_fit_x_generator():
```

Comment:
I think we're overfitting to the tensorflow case for unit testing argument exemption. In https://github.com/mlflow/mlflow/pull/5683/files#r851466125, I've proposed that we instead define exemptions as allowlist entries. Then, we can mock the allowlist here and verify that arguments are exempted properly for different cases (e.g. try a couple of different functions & types).

This way, we can also test that, e.g. if the type doesn't match or the function name doesn't match or the integration name doesn't match, arguments aren't exempted (negative testing).